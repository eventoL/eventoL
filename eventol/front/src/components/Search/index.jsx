import React from 'react';
import PropTypes from 'prop-types';

import './index.scss';

export default class Search extends React.PureComponent {
  static propTypes = {
    onChange: PropTypes.func,
    onEnter: PropTypes.func.isRequired,
  };

  static defaultProps = {
    onChange: () => {},
  };

  state = {
    searchTerm: '',
  };

  handleKeyUp = ({key}) => {
    if (key === 'Enter') {
      const {onEnter} = this.props;
      const {searchTerm} = this.state;
      onEnter(searchTerm);
    }
  };

  handleChange = ({target: {value: searchTerm}}) => {
    this.setState({searchTerm});
    const {onChange} = this.props;
    onChange(searchTerm);
  };

  render() {
    const {searchTerm} = this.state;
    return (
      <div className="search" id="search">
        <input
          id="search-input"
          onChange={this.handleChange}
          onKeyUp={this.handleKeyUp}
          placeholder={gettext('Search by title...')}
          type="search"
          value={searchTerm}
        />
      </div>
    );
  }
}
