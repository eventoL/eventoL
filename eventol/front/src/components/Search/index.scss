@import '../../styles/variables';
@import '../../styles/commons';

.search {
  @extend %flex-element;
  @extend %center-element;
  left: 0;
  pointer-events: none;
  position: absolute;
  width: 100%;

  input {
    -moz-appearance: none;
    -webkit-appearance: none;
    -webkit-transition: border .25s ease, width .125s ease .125s;
    appearance: none;
    background: transparent;
    border: 2px solid $black;
    border-radius: 37px;
    color: $text-color;
    font-family: 'Lato', sans-serif;
    font-size: 14px;
    font-weight: 300;
    height: 37px;
    outline: none;
    padding: 0 10px;
    pointer-events: all;
    transition: border .25s ease, width .125s ease .125s;
    width: 35vw;
    min-width: 150px;
  }

  input:focus {
    border: 2px solid $eventol-color;
    width: 45vw;
  }
}
