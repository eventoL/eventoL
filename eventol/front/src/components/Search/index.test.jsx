import wait from 'waait';
import React from 'react';
import renderer from 'react-test-renderer';

import Search from '.';

describe('Search', () => {
  let tree;
  let onEnter;
  let element;
  let onChange;
  let instance;
  let component;

  const getComponent = () => {
    element = <Search onChange={onChange} onEnter={onEnter} />;
    component = renderer.create(element);
    return component;
  };

  beforeEach(() => {
    onChange = jest.fn();
    onEnter = jest.fn();
    component = getComponent();
    instance = component.root;
    tree = component.toJSON();
  });

  describe('Default', () => {
    test('Snapshot', () => {
      expect(tree).toMatchSnapshot();
    });

    test('Snapshot with update text', async () => {
      expect(tree).toMatchSnapshot();

      const value = 'searchTerm';
      instance.findByType('input').props.onChange({target: {value}});
      component.update(element);
      await wait(0);

      tree = component.toJSON();
      expect(tree).toMatchSnapshot();
      expect(instance.findByType('input').props.value).toEqual(value);
    });
  });

  test('handle onChange', () => {
    const value = 'searchTerm';
    instance.findByType('input').props.onChange({target: {value}});
    expect(onChange).toBeCalled();
    expect(onChange).toBeCalledWith(value);
  });

  test('handle onChange without onChange function', () => {
    onChange = undefined;
    component = getComponent();
    instance = component.root;

    const value = 'searchTerm';
    instance.findByType('input').props.onChange({target: {value}});
  });

  describe('handle onKeyUp', () => {
    test('With not Enter key', () => {
      instance.findByType('input').props.onKeyUp({key: 'A'});
      expect(onEnter).not.toBeCalled();
    });

    test('With Enter key', () => {
      instance.findByType('input').props.onKeyUp({key: 'Enter'});
      expect(onEnter).toBeCalled();
    });
  });
});
