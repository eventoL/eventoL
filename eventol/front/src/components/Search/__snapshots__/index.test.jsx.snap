// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Default Snapshot 1`] = `
<div
  className="search"
  id="search"
>
  <input
    id="search-input"
    onChange={[Function]}
    onKeyUp={[Function]}
    placeholder="Search by title…"
    type="search"
    value=""
  />
</div>
`;

exports[`Search Default Snapshot with update text 1`] = `
<div
  className="search"
  id="search"
>
  <input
    id="search-input"
    onChange={[Function]}
    onKeyUp={[Function]}
    placeholder="Search by title…"
    type="search"
    value=""
  />
</div>
`;

exports[`Search Default Snapshot with update text 2`] = `
<div
  className="search"
  id="search"
>
  <input
    id="search-input"
    onChange={[Function]}
    onKeyUp={[Function]}
    placeholder="Search by title…"
    type="search"
    value="searchTerm"
  />
</div>
`;
